backend/controllers/addAppointmentDetails.js

```diff
@@ -23,6 +23,67 @@ const setSelected = (value, items) =>
         : item
   )
 
+const buildDateTime = ({ date, hours, minutes }) => {
+  const time = date && hours && minutes && moment(date, DAY_MONTH_YEAR)
+  return time && time.hour(Number(hours)).minutes(Number(minutes))
+}
+
+const extractHoursMinutes = dateTime => {
+  const instant = moment(dateTime, DATE_TIME_FORMAT_SPEC)
```

Comment:
(see below comments) so think here you could cope with not being set by doing something like:
```javascript
if (!dateTime) return { hours: '', minutes: ''}
```